libs/chain-signatures/contract/src/tee/tee_state.rs

```diff
@@ -30,31 +31,59 @@ impl Default for TeeState {
             allowed_docker_image_hashes: Default::default(),
             historical_docker_image_hashes: Default::default(),
             votes: Default::default(),
+
             tee_participant_info: IterableMap::new(StorageKey::TeeParticipantInfo),
         }
     }
 }
-
 impl TeeState {
+    /// A shared helper method to verify the TEE quote and Docker image.
+    /// This method returns a `Result` with the `TeeQuoteStatus` or an `Error`.
+    pub fn verify_tee_participant(
+        &mut self,
+        tee_participant_info: &TeeParticipantInfo,
+        sign_pk: &PublicKey,
+        timestamp_s: u64,
+    ) -> Result<TeeQuoteStatus, Error> {
+        let quote_result = tee_participant_info.verify_quote(timestamp_s);
+
+        if let Ok(verified_report) = quote_result {
+            let allowed = self.get_allowed_hashes();
+            let historical = self.get_historical_hashes();
+            let docker_valid = tee_participant_info
+                .verify_docker_image(&allowed, &historical, verified_report, sign_pk.clone())
+                .unwrap_or(false);
```

Comment:
Shouldn't we propagate the error here instead of silently returning false?

```suggestion
                .verify_docker_image(&allowed, &historical, verified_report, sign_pk.clone())?;
```

Reply:
https://github.com/near/mpc/pull/816#discussion_r2272437257